test/test_contrib/test_markdown_renderer.py

```diff
@@ -0,0 +1,147 @@
+from unittest import TestCase
+
+from contrib.markdown_renderer import MarkdownRenderer
+from mistletoe.block_token import Document
+
+
+class TestMarkdownRenderer(TestCase):
+    @staticmethod
+    def roundtrip(input):
+        with MarkdownRenderer() as renderer:
+            rendered = renderer.render(Document(input))
+            return rendered
+
+    def test_empty_document(self):
+        input = []
+        rendered = self.roundtrip(input)
+        self.assertEqual(rendered, "".join(input))
+
+    def test_paragraphs_and_blank_lines(self):
+        input = ['Paragraph 1. Single line. Followed by two white-space-only lines.\n',
```

Comment:
Tip: for these multi-line inputs, it is probably easier and good for readability to use Python's multi-line strings, together with `dedent()` if we want to keep test code indented - see https://github.com/miyuchina/mistletoe/blob/master/test/test_traverse.py#L28 as an example.

Reply:
Yeah, I started out like that, but switched to this format because that's how most of the other test cases in the project were written. Turns out that this format was more readable (at least in my eyes). Especially for test cases like the code blocks where indentation is very irregular -- and important.

Reply:
Yeah, it seems like a matter of taste... I pretty much like the multi-line format, because it is less typing and generally faster for reading (at least for me ;)). But of course, at minimum for the sake of consistency with most of the existing tests, your choice is just OK as well. :)